src/Features/LanguageServer/Protocol/Features/CodeFixes/CodeFixService.cs

```diff
@@ -241,6 +292,28 @@ internal partial class CodeFixService : ICodeFixService
             }
         }
 
+        private static async Task<ImmutableArray<DiagnosticData>> GetCopilotDiagnosticsAsync(
+            TextDocument document,
+            TextSpan range,
+            CodeActionRequestPriority? priority,
+            CancellationToken cancellationToken)
+        {
+            if (!(priority is null or CodeActionRequestPriority.Low)
+                || document is not Document sourceDocument)
+            {
+                return [];
+            }
+
+            // Expand the fixable range for Copilot diagnostics to containing method.
+            // TODO: Share the below code with other places we compute containing method for Copilot analysis.
+            var root = await sourceDocument.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var syntaxFacts = sourceDocument.GetRequiredLanguageService<ISyntaxFactsService>();
+            var containingMethod = syntaxFacts.GetContainingMethodDeclaration(root, range.Start, useFullSpan: false);
```

Comment:
this is the the third time i've seen this logic in separate places, without explanation.  It looks to be a critical part of how all these components agree on where they're going to operate (or something).  Please consolidate in some service htey all use.